core/src/main/java/org/elasticsearch/common/ESMessageFormat.java

```diff
@@ -17,46 +17,118 @@
  * under the License.
  */
 
-package org.elasticsearch.common.logging.support;
+package org.elasticsearch.common;
 
 import java.util.HashMap;
 import java.util.Map;
 
 /**
+ * A message format in elasticsearch. Supports the following features:
+ *<pre>
  *
+ * - `{}`   indicates a place holder where single values will be translated to a string ({@code null}
+ *          will be translated to a {@code "null"} string). Arrays will be enclosed in brackets (`[` and `]`).
+ *
+ * - `[]`   indicates a place holder where single values will be translated to a string enclosed in
+ *          brackets `[` and `]`. Arrays will also be enclosed in brackets (but not double ones)
+ *
+ * examples:
+ *
+ *      PATTERN                 VALUE           OUTPUT
+ *
+ *      "value {}"              2               "value 2"
+ *      "value {}"              null            "value null"
+ *      "value []"              "text"          "value [text]"
+ *      "value []"              null            "value [null]"
+ *      "values {}"             [1, 2]          "values [1, 2]"
+ *      "values []"             [1, 2]          "values [1, 2]"
+ * </pre>
  */
-public class LoggerMessageFormat {
+public class ESMessageFormat {
```

Comment:
What about using the long version `ElasticsearchMessageFormat` like we already do for exceptions, tests, etc. ?